### DISCUSSION TITLE
Supporting `eq` on nested tables fields.

### DISCUSSION THREAD

**Discussion by dbrrt** [supabase/supabase#12688] [Questions] [answered]
When querying data via the Supabase JS client library, it's possible to join on other tables. However, it's not possible to query leveraging the join between tables.

For instance doing something like this: 
```typescript
  const { data } = await supabase
    .from("accounts_users")
    .select(
      `*,
    name: accounts_users_name(formatted)
    `
    )
    .eq("name.formatted", "John Lennon");
```

**Comment by GaryAustin1** [comment]
Sure:
![image](https://user-images.githubusercontent.com/54564956/221419576-7e58f977-63da-4175-b460-fbcb6a80d123.png)

**Reply by dbrrt** [reply]
Thanks for your answer, I missed that from the docs

**Reply by dbrrt** [reply]
@GaryAustin1 This doesn't exclude the result from the upper collection.
in your case, it'll return cities names even if the country doesn't match.

**Reply by GaryAustin1** [reply, marked as answer]
You need to add !inner to the select like `countries!inner(*)`  See the note in the image I posted.
